### DISCUSSION TITLE
Welcome to td-ameritrade-python-api Discussions!

### DISCUSSION THREAD

**Discussion by areed1192** [areed1192/td-ameritrade-python-api#171] [Ideas]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by areed1192** [comment]
## Refactoring

Howdy everyone, so I've been looking over the library and I wanted to gauge what the interest would be for a reorganization of the library. Right now, all the endpoints are stored in a single file called `client.py`. My goal would be to break that file into smaller files that would focus on specific services.

For example, `accounts.py` would contain all the account endpoints and would be accessible from the `AccountServices` object. If you're having a tough time grasping this, I would say look at one of my other libraries for the [Power BI Rest API](https://github.com/areed1192/power-bi-python-api). That should give you an idea of where I'm trying to take this.

What is everyone's thought on this? Do we think it would make the library easier to use/more resilient or are we just going down the wrong path?

## OAuth

Also, I'm getting a lot of mixed reviews on the `OAuth` process. Some people like it and others seemed to find it more of obstacle than a benefit. What I was going to propose is to create a new file to handle the `OAuth` process but give multiple methods to handle `OAuth`. For example, you could provide the token yourself, you could provide the file, and so on.

I would also like to get feedback on that and hear what people are thinking.

**Reply by amn-sharma** [reply]
I think this is a good idea. Also, consolidating other libraries like trading-robot and building a robust program will be a great asset. 
Let me know if any help is needed in refactoring. Thank You for your work.

**Reply by areed1192** [reply]
Yeah, it's sounding like most everyone agrees that the library could use some refactoring/reorganization. Now out of curiosity, would you want to leave this repo as is or have a new repo made which would house the refactored version?

**Comment by dylkimrik** [comment]
A new repo would be great. IMO. I find it enjoyable to go back and see the
previous work

On Sat, Apr 3, 2021 at 1:35 PM Alex Reed ***@***.***> wrote:

> Yeah, it's sounding like most everyone agrees that the library could use
> some refactoring/reorganization. Now out of curiosity, would you want to
> leave this repo as is or have a new repo made which would house the
> refactored version?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/areed1192/td-ameritrade-python-api/discussions/171#discussioncomment-566193>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APUIRS6LPV5DG2BM2BOTBELTG5GUZANCNFSM42JDJGAQ>
> .
>
-- 
Dylan R Kimmons-Ruikka
Cell: (727) 310 7773
Email: ***@***.***

**Reply by areed1192** [reply]
Now comes the fun part, thinking of a new name :)

**Comment by zwopr** [comment]
I'm unable to get FX price_history (EUR/USD).  It works with equities.  Is it just me?

json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
